src/main/java/com/williamfiset/algorithms/datastructures/queue/Queue.java

```diff
@@ -1,53 +1,17 @@
-/**
- * A simple queue implementation with a linkedlist
- *
- * @author William Fiset, william.alexandre.fiset@gmail.com
- */
 package com.williamfiset.algorithms.datastructures.queue;
 
-public class Queue<T> implements Iterable<T> {
-
-  private java.util.LinkedList<T> list = new java.util.LinkedList<T>();
-
-  public Queue() {}
-
-  public Queue(T firstElem) {
-    offer(firstElem);
-  }
-
-  // Return the size of the queue
-  public int size() {
-    return list.size();
-  }
-
-  // Returns whether or not the queue is empty
-  public boolean isEmpty() {
-    return size() == 0;
-  }
+/**
+ * @author liujingkun, liujkon@gmail.com
+ * @param <T> the type of element held int the queue
+ */
+public interface Queue<T> {
```

Comment:
Unlike the Stack, Java actually already has a Queue interface:
https://docs.oracle.com/javase/7/docs/api/java/util/Queue.html

Please use the java.util.Queue instead

Reply:
java.util.Queue extends from the Collection interface. if use java.util.Queue, we have to implement lots of methods defined in the Collection interface. it seems to be messed.

Reply:
> Unlike the Stack, Java actually already has a Queue interface:
> https://docs.oracle.com/javase/7/docs/api/java/util/Queue.html
> 
> Please use the java.util.Queue instead

java.util.Queue extends from the Collection interface. if use java.util.Queue, we have to implement lots of methods defined in the Collection interface. it seems to be messed.

Reply:
Ok, it's fine then.